### DISCUSSION TITLE
Are alternative scripting tools allowed for Oh-My-Zsh plugins?

### DISCUSSION THREAD

**Discussion by 2art** [ohmyzsh/ohmyzsh#12514] [Help] [answered]
I was wondering, am I allowed to use something like Python for parts of a slightly bigger plugin project, or does it all have to be native Zsh code?

**Comment by KianAnbarestani** [comment, marked as answer]
@2art 

Yes, you're allowed to use Python for parts of your plugin. There's no strict requirement that everything must be written in native Zsh. However, keep in mind that using Python will introduce an external dependency, so document it and ensure it's available on target systems. Integrate your Python scripts cleanly with your Zsh code by calling them as needed, while providing fallbacks or error messages if Python isn’t installed.

**Reply by 2art** [reply]
Thank you for your reply, although I had forgotten this issue already and focused on pure Zsh; But given this information, I will be looking to implement more complicated tasks with alternative tools.

**Comment by roeniss** [comment]
[ssh-proxy](https://github.com/ohmyzsh/ohmyzsh/blob/53c4c4671a931d9f7a3b39fe9131c6f84fe27ca5/plugins/shell-proxy/ssh-proxy.py) plugin is one of examples.
